### PR TITLE
Fix remote access whitelist

### PR DESCRIPTION
**Pull request by BaronGreenback**
Bug is a logic inversion.

Moved the code into the networkManager so that a test function could be added.

### CODE REVIEW

File: tests/Jellyfin.Networking.Tests/NetworkParseTests.cs [hunk 1]
```diff
@@ -514,5 +514,25 @@ public void TestBindInterfaceOverrides(string source, string lan, string bindAdd
 
             Assert.Equal(intf, result);
         }
+
+        [Theory]
+        [InlineData("185.10.10.10,200.200.200.200", "79.2.3.4", false, true)] // whitelist
+        [InlineData("185.10.10.10", "185.10.10.10", false, false)] // whitelist
+        [InlineData("185.10.10.10", "79.2.3.4", true, false)] // blacklist
+        public void TestRemoteAccess(string addresses, string remoteIp, bool blacklist, bool denied)
```
Comment (cvium) [inline tests/Jellyfin.Networking.Tests/NetworkParseTests.cs:522]: I know it matches the naming convention in the file but I'm trying to get us to use `MethodName_Scenario_ExpectedBehaviour` eg. `HasRemoteAccess_GivenNonEmptyWhitelist_AllowsOnlyIpsInWhitelist`. I would also like for this to be split into two tests, one for blacklist and one for whitelist.

File: Jellyfin.Networking/Manager/NetworkManager.cs [hunk 2]
```diff
@@ -576,6 +576,46 @@ public bool TryParseInterface(string token, out Collection<IPObject>? result)
...
+                    {
+                        // Black list, so flip over.
+                        isListed = !isListed;
+                    }
+
+                    if (!isListed)
+                    {
+                        // If your name isn't on the list, you arn't coming in.
+                        return false;
+                    }
```
Comment (cvium) [inline Jellyfin.Networking/Manager/NetworkManager.cs:607]: This should be the same I think. Not sure if it's more confusing though.
```suggestion
                    return RemoteAddressFilter.ContainsAddress(remoteIp) == !config.IsRemoteIPFilterBlacklist;
```

File: Jellyfin.Networking/Manager/NetworkManager.cs [hunk 3]
```diff
@@ -576,6 +576,46 @@ public bool TryParseInterface(string token, out Collection<IPObject>? result)
...
+        /// <returns><b>True</b> if has access, otherwise <b>false</b>.</returns>
+        public bool HasRemoteAccess(IPAddress remoteIp)
+        {
+            var config = _configurationManager.GetNetworkConfiguration();
+            if (config.EnableRemoteAccess)
+            {
+                // Comma separated list of IP addresses or IP/netmask entries for networks that will be allowed to connect remotely.
+                // If left blank, all remote addresses will be allowed.
+                var remoteAddressFilter = RemoteAddressFilter;
+
```
Comment (cvium) [inline Jellyfin.Networking/Manager/NetworkManager.cs:592]:
```suggestion
```

**Review by cvium — CHANGES_REQUESTED** [review]

File: Jellyfin.Networking/Manager/NetworkManager.cs (hunk 2 above)
Comment (BaronGreenback) [reply Jellyfin.Networking/Manager/NetworkManager.cs:607]: Can't be any more confusing than the original. Don't know how many time's i've scratched my head with it.

File: tests/Jellyfin.Networking.Tests/NetworkParseTests.cs [hunk 4]
```diff
@@ -514,5 +514,42 @@ public void TestBindInterfaceOverrides(string source, string lan, string bindAdd
...
+            };
+            using var nm = new NetworkManager(GetMockConfig(conf), new NullLogger<NetworkManager>());
+
+            Assert.NotEqual(nm.HasRemoteAccess(IPAddress.Parse(remoteIp)), denied);
+        }
+
+        [Theory]
+        [InlineData("185.10.10.10", "79.2.3.4", false)] // blacklist
+        [InlineData("185.10.10.10", "185.10.10.10", true)] // blacklist
+        public void HasRemoteAccess_GivenNonEmptBlacklist_BlacklistTheIps(string addresses, string remoteIp, bool denied)
```
Comment (cvium) [inline tests/Jellyfin.Networking.Tests/NetworkParseTests.cs:539]:
```suggestion
        public void HasRemoteAccess_GivenNonEmptyBlacklist_BlacklistTheIps(string addresses, string remoteIp, bool denied)
```

File: Jellyfin.Networking/Manager/NetworkManager.cs [hunk 5]
```diff
@@ -576,6 +576,33 @@ public bool TryParseInterface(string token, out Collection<IPObject>? result)
             return false;
         }
 
+        /// <summary>
+        /// Checks to see if <paramref name="remoteIp"/> has access.
+        /// </summary>
+        /// <param name="remoteIp">IP Address of client.</param>
+        /// <returns><b>True</b> if has access, otherwise <b>false</b>.</returns>
```
Comment (barronpm) [inline Jellyfin.Networking/Manager/NetworkManager.cs:583]: No need to duplicate this, `inheritdoc` will work

**Review by barronpm — APPROVED** [review]

File: tests/Jellyfin.Networking.Tests/NetworkParseTests.cs [hunk 6]
```diff
@@ -514,5 +514,42 @@ public void TestBindInterfaceOverrides(string source, string lan, string bindAdd
...
+                IsRemoteIPFilterBlacklist = false
+            };
+            using var nm = new NetworkManager(GetMockConfig(conf), new NullLogger<NetworkManager>());
+
+            Assert.NotEqual(nm.HasRemoteAccess(IPAddress.Parse(remoteIp)), denied);
+        }
+
+        [Theory]
+        [InlineData("185.10.10.10", "79.2.3.4", false)] // blacklist
+        [InlineData("185.10.10.10", "185.10.10.10", true)] // blacklist
```
Comment (cvium) [inline tests/Jellyfin.Networking.Tests/NetworkParseTests.cs:538]: Maybe include tests for empty lists?

Comment (BaronGreenback) [reply tests/Jellyfin.Networking.Tests/NetworkParseTests.cs]: done

**Review by cvium — APPROVED** [review]

File: tests/Jellyfin.Networking.Tests/NetworkParseTests.cs [hunk 7]
```diff
@@ -514,5 +514,45 @@ public void TestBindInterfaceOverrides(string source, string lan, string bindAdd
...
+        [InlineData("", "100.100.100.100", false)]
+
+        public void HasRemoteAccess_GivenWhitelist_AllowsOnlyIpsInWhitelist(string addresses, string remoteIp, bool denied)
+        {
+            // Comma separated list of IP addresses or IP/netmask entries for networks that will be allowed to connect remotely.
+            // If left blank, all remote addresses will be allowed.
+            var conf = new NetworkConfiguration()
+            {
+                EnableIPV4 = true,
+                RemoteIPFilter = addresses.Split(","),
```
Comment (Bond-009) [inline tests/Jellyfin.Networking.Tests/NetworkParseTests.cs:530]:
```suggestion
                RemoteIPFilter = addresses.Split(','),
```
It's only in the tests, but still

File: tests/Jellyfin.Networking.Tests/NetworkParseTests.cs [hunk 8]
```diff
@@ -514,5 +514,45 @@ public void TestBindInterfaceOverrides(string source, string lan, string bindAdd
...
+        [InlineData("185.10.10.10", "185.10.10.10", true)]
+        [InlineData("", "100.100.100.100", false)]
+        public void HasRemoteAccess_GivenBlacklist_BlacklistTheIps(string addresses, string remoteIp, bool denied)
+        {
+            // Comma separated list of IP addresses or IP/netmask entries for networks that will be allowed to connect remotely.
+            // If left blank, all remote addresses will be allowed.
+            var conf = new NetworkConfiguration()
+            {
+                EnableIPV4 = true,
+                RemoteIPFilter = addresses.Split(","),
```
Comment (Bond-009) [inline tests/Jellyfin.Networking.Tests/NetworkParseTests.cs:549]:
```suggestion
                RemoteIPFilter = addresses.Split(','),
```

**Review by Bond-009 — APPROVED** [review]

File: tests/Jellyfin.Networking.Tests/NetworkParseTests.cs (hunk 7 above)
Comment (BaronGreenback) [reply tests/Jellyfin.Networking.Tests/NetworkParseTests.cs]: done all 30
